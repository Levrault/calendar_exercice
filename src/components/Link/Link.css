.link {
  background: none;
  color: $palette-primary-text;
  position: relative;
  text-align: center;
  text-decoration: none;
  width: resolve($space-xl + $space-l);
  z-index: 20;

  &:visited {
    text-decoration: none;
  }
  
  &:link {
    text-decoration: none;
  }

  &:active {
    text-decoration: none;
  }

  &::after {
    background: $palette-accent;
    bottom: -24px;
    content: '';
    height: 1px;
    left: 50%;
    position: absolute;
    right: 50%;
    transition: all ease .5s;
  } 

  &.active,
  &:hover {
    color: $palette-accent;

    &::after {
      height: $space-xs;
      left: 0;
      right: 0;
    }
  } 

  &::before {
    font-size: 8px;
    left: -10px;
    opacity: .3;
    position: absolute;
    transform: translateX(-100%);
  }
}
