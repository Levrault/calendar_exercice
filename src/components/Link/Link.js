import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import styles from './Link.css';

/**
 * Generate Navlink with proper styles
 * @param {object} props
 * @returns {node}
 */
const Link = (props) => (
  <NavLink {...props} exact activeClassName={styles.active} />
);

Link.propstype = {
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired
};

Link.defaultProps = {
  className: styles.link
};

export default Link;
