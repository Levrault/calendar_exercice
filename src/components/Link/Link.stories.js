import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import { withMarkdownNotes } from '@storybook/addon-notes';
import StoryRouter from 'storybook-react-router';
import Wrapper from '../../tools/storybook/Wrapper';
import Link from './Link';

storiesOf('Link', module)
  .addDecorator(withKnobs)
  .addDecorator(StoryRouter())
  .addDecorator(Wrapper)
  .add('default', withMarkdownNotes(`
    # Link

    use react-router component to manage link
    `)(() => (
    <Link
      title={text('title', 'link')}
      to={text('to', 'to')}
    >
      Link
    </Link>
  )));
