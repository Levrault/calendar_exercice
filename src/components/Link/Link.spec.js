import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

jest.mock('react-router-dom', () => ({
  Link: jest.fn(({ title, children }) =>
    <a title={title}>This is a react-router:Link component: {children}</a>,
  )
}));

describe('Link component', () => {
  it('should render without problem', () => {
    const component = renderer.create(
      <Link title="Title" to="link" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
