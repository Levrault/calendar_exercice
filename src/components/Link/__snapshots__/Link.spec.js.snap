// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should render without problem 1`] = `
<a
  title="Title"
>
  This is a react-router:Link component: 
</a>
`;
